examples/DallasTemperatureSensor/DallasTemperatureSensor.ino

```diff
@@ -22,88 +22,179 @@
+float lastTemperature[maxAttachedDS18B20];        // creates an array to hold the previous temperature measurements for each possible sensor.
+int numSensors=0;                                 // variable to contain the number of found attached sensors.
+unsigned long measurementSleepTime = 0;           // variable to store the calculated Sleep time if the node is battery powered.
+
+
+// Mysensors settings
+MyMessage msg(0,V_TEMP);                          // Sets up the message format that we'l be sending to the MySensors gateway later.
+
 
 void before()
 {
-  // Startup up the OneWire library
-  sensors.begin();
+   sensors.begin();                               // Startup up the OneWire library. It allows multiple sensors to talk over one wire (one pin).
 }
 
-void setup()  
-{ 
-  // requestTemperatures() will not block current thread
-  sensors.setWaitForConversion(false);
+void setup()
+{
+   for(int i=0; i<maxAttachedDS18B20; i++) 
+   {
+      lastTemperature[i] = 0;  //Pre-filling array with 0's.
+   }
+   sensors.setWaitForConversion(false); // requestTemperatures() will not block current thread
+
+#ifdef BATTERY_POWERED // If batterypowered, we'll let Sleep take over the scheduling.
+   measurementSleepTime = measurementInterval;
+   measurementInterval = 1; // When the Arduino is asleep, millis doesn't increment anymore (time stops as it were). To fix this, we'll set the measurement interval time to 1, so that when the arduino wakes up it will immediately try to measure again.
+#endif
+
+   Serial.begin(115200); // for serial debugging.
+   Serial.print("Hello world, I am a sensor. \n");
 }
 
-void presentation() {
-  // Send the sketch version information to the gateway and Controller
-  sendSketchInfo("Temperature Sensor", "1.1");
+void presentation()
+{
+   sendSketchInfo("Temperature Sensor", "1.2");    // Send the sketch version information to the gateway and Controller
+   numSensors = sensors.getDeviceCount();          // Fetch the number of attached temperature sensors
+   for (int i=0; i<numSensors && i<maxAttachedDS18B20; i++) 
+   {
+      present(i, S_TEMP);                          // Present all sensors to controller (16 maximum).
+   }
+}
 
-  // Fetch the number of attached temperature sensors  
-  numSensors = sensors.getDeviceCount();
 
-  // Present all sensors to controller
-  for (int i=0; i<numSensors && i<MAX_ATTACHED_DS18B20; i++) {   
-     present(i, S_TEMP);
-  }
+void loop()
+{
+
+   // You should not change these variables:
+   static boolean isMeasuring = true;                        // Used to indicate when the time is right for a new measurement to be made.
+   static boolean isCalculating = false;                     // Used to bridge the time that is needed to calculate the temperature values by the Dallas library.
+   static unsigned long currentMillis = 0;                   // The millisecond clock in the main loop.
```

Comment:
No need to make this static as it will be initialized to millis() a few lines down.